### PR TITLE
Add GALAH dataset

### PR DESCRIPTION
**Pull request by maja-jablonska**

### CODE REVIEW

**Comment by maja-jablonska** [comment]
I'd say it's almost done, I'm just not sure how to process resolution. I asked at ANU.

**Comment by maja-jablonska** [comment]
I'm thinking of providing some util functions, e.g. to extract the spectra fluxes corresponding to three arms of the spectrograph - it might be a bit confusing to see a combined dataset like this.

image

**Comment by maja-jablonska** [comment]
Added filter indices so that it can be viewed per-arm!

image

**Review by mb010 — CHANGES_REQUESTED** [review]
Can we remove the cached data from the ipython notebook cells?

**Review by mb010 — CHANGES_REQUESTED** [review]
@maja-jablonska if you have a chance to update the PR this week, that would be ideal. If not, let me know and I'll have a look.

File: scripts/galah/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - snr_c3_iraf > 30           # A recommended minimum signal-to-noise ratio
+    - flag_sp = 0    
+    - flag_fe_h = 0
+```
+
+These cuts follow the ![GALAH DR3 best practices](https://www.galah-survey.org/dr3/using_the_data/).
+
+
+## Data preparation
+
```
Comment (mb010) [inline scripts/galah/README.md:20]:
```suggestion
The data can be loaded for example as follows:
```python
from datasets import load_dataset
import matplotlib.pyplot as plt

dataset = load_dataset('./galah.py', trust_remote_code=True, split='train')
spectrum = dataset['train'][0]['spectrum']
spectrum.keys()

filter_ = 'B'
s_ind, e_ind = dataset['train'][0]['filter_indices'][f'{filter_}_start'], dataset['train'][0]['filter_indices'][f'{filter_}_end']
plt.plot(spectrum['lambda'][s_ind:e_ind], spectrum['flux'][s_ind:e_ind], color='royalblue')
plt.xlabel(r'wavelength ($\AA$)')
plt.ylabel(r'flux (erg/s/cm$^2$/$\AA$)')
plt.title('Spectrum for Selected Star: B Filter')
```
```

**Review by mb010 — APPROVED** [review]
lgtm

**Comment by mb010** [comment]
GALAH completed fine. Apogee failed in the build test with the outdated configuration. Going forward with merge.

**Pull request merged**